Bluetooth: host: reword scary warnings

jori-nordic opened the pull request:
This PR downgrades a few stack warnings (to debug level) that should only really be seen by the stack developers and not the end-users.

jori-nordic committed:
Bluetooth: hci_core: reword warning

Reword the scary-looking `bt_hci_core; Malformed data` warning to show
where it is originating from.

Signed-off-by: Jonathan Rico <jonathan.rico@nordicsemi.no>

jori-nordic committed:
Bluetooth: smp: reword pairing failed error

This makes it more user-friendly, in case no failed cb has been
registered so this error is printed out of context.

Signed-off-by: Jonathan Rico <jonathan.rico@nordicsemi.no>

jori-nordic committed:
Bluetooth: l2cap: reword unknown pdu warning

This is a scary-looking warning for users, and should only really matter to
stack developers. It will eventually show up in the wild for embedded
devices that don't get updated to the latest and greatest.

Reword it to make it clear we are rejecting it, and it doesn't impact the
stack in any way.

Signed-off-by: Jonathan Rico <jonathan.rico@nordicsemi.no>

jori-nordic committed:
Bluetooth: att: precise warning messages

- Downgrade the warning emitted when we run out of channels.
- Only log a warning when failing to establish channels and the error is
  not -ENOMEM.
- Add a debug log message in that case to hint what might've happened.

This should make it less confusing to the average user not aware of the
stack internals.

Signed-off-by: Jonathan Rico <jonathan.rico@nordicsemi.no>

jhedberg commented on subsys/bluetooth/host/hci_core.c:
```diff
@@ -4020,7 +4020,7 @@ void bt_data_parse(struct net_buf_simple *ad,
 
 		if (len > ad->len) {
-			BT_WARN("Malformed data");
+			BT_DBG("malformed data");
```
`bt_data_parse()` doesn't do any error reporting to the caller, so we'd be potentially suppressing important information to the application developer with the downgrade from WARN to DBG.

The general principle that we've tried to follow in the stack is that we use BT_WARN() whenever we receive something from a peer that we consider invalid or that we don't support, but which we can nevertheless handle gracefully (ignore or send an error PDU response) BT_ERR() on the other hand is used for local failures. What we can do, instead of changing the log level, is to better word the content of these log messages, to make it clear what they're about and that way avoid worrying the app developer. Btw, this comment applies to all other places as well in this PR where you're downgrading WARN to DBG for invalid or unsupported peer data. 

As for the placement of the implementation of `bt_data_parse()` I'm fine with moving it to a better place. Most likely it went into `hci_core.c` originally, since there wasn't any other more suitable "utilities" module back then (is there now?)

jhedberg commented on subsys/bluetooth/host/att.c:
```diff
@@ -3257,7 +3257,11 @@ static void att_enhanced_connection_work_handler(struct k_work *work)
 
-	if (err < 0) {
+	if (err == -ENOMEM) {
+		BT_DBG("Failed to connect %d EATT channels, central has probably "
+		       "already established some.",
+		       att->eatt.chans_to_connect);
```
Same here, this is fine

jhedberg commented on subsys/bluetooth/host/att.c:
```diff
@@ -3206,7 +3206,7 @@ static struct bt_att_chan *att_chan_new(struct bt_att *att, atomic_val_t flags)
 
 		if (quota == ATT_CHAN_MAX) {
-			BT_WARN("Maximum number of channels reached: %d", quota);
+			BT_DBG("Maximum number of channels reached: %d", quota);
```
DBG() is fine in this case.

Thalley commented on subsys/bluetooth/host/hci_core.c:
```diff
@@ -4020,7 +4020,7 @@ void bt_data_parse(struct net_buf_simple *ad,
 
 		if (len > ad->len) {
-			BT_WARN("Malformed data");
+			BT_WARN("%s: malformed advertising data", __func__);
```
Why the addition of the function name? 

If a user wants the function name in the warning logs, they can just enable `CONFIG_LOG_FUNC_NAME_PREFIX_WRN`

jori-nordic replied:
well, maybe that symbol should be selected by `CONFIG_BT_DEBUG_LOG`. I can make another commit to do so if you agree.
Having app developers needing to grep for log messages is not very nice IMO.

Thalley replied:
> well, maybe that symbol should be selected by `CONFIG_BT_DEBUG_LOG`. I can make another commit to do so if you agree. Having app developers needing to grep for log messages is not very nice IMO.

Yeah, honestly I do not know why `CONFIG_LOG_FUNC_NAME_PREFIX_WRN` and `CONFIG_LOG_FUNC_NAME_PREFIX_ERR` for that matter is not selected when debugging. In many cases the name of the function for errors and warnings are just as, or even more, useful than the names of the functions for debug log. 

I wouldn't mind if those were selected when doing debug logs :)

jori-nordic committed:
Bluetooth: log: show function names by default

Show function names for all BT_xx logging levels when BT_DEBUG_LOG is
enabled. This makes it easier to find the problematic lines.

Signed-off-by: Jonathan Rico <jonathan.rico@nordicsemi.no>
